### DISCUSSION TITLE
Attachment

### DISCUSSION THREAD

**Discussion by edmaSB** [orchidsoftware/platform#2870] [Q&A]
Hi, I have a problem with attachments linked to a template. I'm using the following code for the form

`public function layout(): iterable
    {
        $machine = new Machine();

        return [
            Layout::rows([

                Input::make('maintenance.employees_id')
                    ->value(auth()->user()->employee->id)
                    ->hidden(true),    
                Input::make('maintenance.maintenance_states_id')
                    ->value(1)
                    ->hidden(true),    
                Select::make('maintenance.maintenance_types_id')
                    ->required()
                    ->fromQuery(MaintenanceType::orderBy('nome', 'asc'), 'nome')
                    ->autoComplete()
                    ->empty(__('Seleziona il tipo'))
                    ->title(__('Tipo di richiesta')),
                Select::make('maintenance.machines_id')
                    ->required()
                    ->fromQuery(Machine::orderBy('nome', 'asc'), 'nome')
                    ->autoComplete()
                    ->empty(__('Seleziona il macchinario'))
                    ->title(__('macchinario')),
                Input::make('maintenance.titolo')
                    //->type('number')
                    ->required()
                    ->title(__('Titolo')),
                TextArea::make('maintenance.descrizione')
                    //->type('number')
                    ->rows(10)
                    ->required()
                    ->title(__('Descrizione')),

            ]),
            Layout::rows([

                Upload::make('maintenance.attachment')
                    ->acceptedFiles('image/*,application/pdf')
                    ->maxFileSize(2)
                    ->maxFiles(5)
                    ->storage('private')
                //->media()
                ,

            ])->title(__('Allegati')),

            BottomSaveButtons::render(),
        ];
    }`

the model is the following

`<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;
use Illuminate\Validation\Rule;
use Orchid\Access\RoleAccess;
use Orchid\Attachment\Attachable;
use Orchid\Attachment\Models\Attachment;
use Orchid\Filters\Filterable;
use Orchid\Filters\Types\Like;
use Orchid\Filters\Types\Where;
use Orchid\Screen\AsSource;

class Maintenance extends Model
{
    use HasFactory, AsSource, Filterable, RoleAccess, Attachable; 

    protected $fillable = [
        'machines_id',
        'employees_id',
        'maintenance_types_id',
        'maintenance_states_id',
        //'skills',
        'titolo',
        'descrizione',
        'preventivo',
    ];

    /**
     * The attributes for which you can use filters in url.
     *
     * @var array
     */
    protected $allowedFilters = [
        'id'         => Where::class,
        'machines_id'         => Where::class,
        'maintenance_types_id'         => Where::class,
        'employees_id'         => Where::class,
        'maintenance_states_id'         => Where::class,
        'titolo'       => Like::class,
        'descrizione'       => Like::class,
        'preventivo'       => Like::class,
    ];

    /**
     * The attributes for which can use sort in url.
     *
     * @var array
     */
    protected $allowedSorts = [
        'id',
        'machines_id',
        'employees_id',
        'maintenance_types_id',
        'maintenance_states_id',
        //'skills',
        'titolo',
        'descrizione',
        'preventivo',
    ];

    public function rules(Maintenance $model)
    {

        return [
            'maintenance.machines_id' => [
                ////'required',
                'required',
                //'unique:skills,slug',
                'integer',
                'exists:machines,id',
                //'max:255',

                //'max:255',
            ],
            'maintenance.employees_id' => [
                ////'required',
                'required',
                //'unique:skills,slug',
                'integer',
                'exists:employees,id',
                //'max:255',

                //'max:255',
            ],
            'maintenance.maintenance_states_id' => [
                ////'required',
                'required',
                //'unique:skills,slug',
                'integer',
                'exists:maintenance_states,id',
                //'max:255',

                //'max:255',
            ],
            'maintenance.maintenance_types_id' => [
                ////'required',
                'required',
                //'unique:skills,slug',
                'integer',
                'exists:maintenance_types,id',
                //'max:255',

                //'max:255',
            ],
            'maintenance.titolo' => [
                'required',
                'string',
                //Rule::unique(Machine::class, 'nome')->ignore($model),
                'max:255',

                //'max:255',
            ],
            'maintenance.descrizione' => [
                'required',
                'string',
                //Rule::unique(Machine::class, 'nome')->ignore($model),
                //'max:255',

                //'max:255',
            ],
            'maintenance.preventivo' => [
                ////'required',
                //'required',
                //'unique:skills,slug',
                'decimal:0,2',
                //'exists:frequencies,id',
                //'max:255',

                //'max:255',
            ],

        ];
    }

    public function machine()
    {
        return $this->hasOne(Machine::class, 'id', 'machines_id');
    }
    public function employee()
    {
        return $this->hasOne(Employee::class, 'id', 'employees_id');
    }
    public function type()
    {
        return $this->hasOne(MaintenanceType::class, 'id', 'maintenance_types_id');
    }
    public function state()
    {
        return $this->hasOne(MaintenanceState::class, 'id', 'maintenance_states_id');
    }

    public function allegati()
    {
        return $this->hasOne(Attachment::class, 'id', 'ticket')->withDefault();
    }
}
`

the files are saved and the attachment table is populated but when I try to recover the files the array is empty.

What am I doing wrong?

**Comment by esmultiservicios** [comment]
I solved this changing Upload for a simple input file, and it works to me
 
     /**
     * Get the fields displayed by the resource.
     *
     * @return array
     */
    public function fields(): array
    {
        return [
            // Grupo de información básica de la empresa
            Group::make([
                Input::make('name')
                    ->title('Nombre')
                    ->id('nameInput')
                    ->placeholder('Ingrese el nombre del cliente')
                    ->type('text')
                    ->autofocus()
                    ->required()
                    ->help('Nombre de la empresa.'),
                Input::make('rtn')
                    ->title('RTN')
                    ->id('rtnInput')
                    ->placeholder('Ingrese el RTN del cliente')
                    ->type('text')
                    ->required()
                    ->help('RTN de la empresa.'),
                Input::make('email')
                    ->title('Correo Electrónico')
                    ->id('emailInput')
                    ->placeholder('Ingrese el correo electrónico de la empresa')
                    ->type('email')
                    ->required()
                    ->help('Correo electrónico de la empresa.'),
            ]),
            // Grupo de información de precios y unidades
            Group::make([
                Input::make('phone')
                    ->title('Teléfono')
                    ->id('phoneInput')
                    ->placeholder('Ingrese el Teléfono')
                    ->type('number')
                    ->step('0.01')  // Permite decimales con dos dígitos después del punto
                    ->required()
                    ->help('Teléfono.'),
                Select::make('unit')
                    ->title('Unidad')
                    ->id('unitSelect')
                    ->options([
                        'kg' => 'Kilogramos',
                        'lb' => 'Libras',
                    ])
                    ->placeholder('Seleccione la unidad')
                    ->required()
                    ->help('Unidad de medida.'),
                Input::make('price')
                    ->title('Precio')
                    ->id('priceInput')
                    ->placeholder('Ingrese el precio')
                    ->type('number')
                    ->step('0.01')  // Permite decimales con dos dígitos después del punto
                    ->required()
                    ->help('Precio.'),
            ]),
            // Grupo de dirección
            Group::make([
                Input::make('upload')
                    ->title('Seleccione una Imágen')
                    ->id('uploadInput')
                    ->placeholder('Seleccione una imágen JPG, JPEG, PNG')
                    ->type('file')
                    ->acceptedFiles('image/jpeg,image/png,image/jpg')
                    ->autofocus(),
                TextArea::make('address')
                    ->title('Dirección')
                    ->id('addressInput')
                    ->rows(5)
                    ->width('100%'),
            ]),
        ];
    }
    
 
 my method is
 
 /**
     * Save the uploaded image and store a copy in the public/uploads folder for the Enterprise model.
     *
     * @param \Illuminate\Http\Request $request
     * @param \App\Models\Enterprise $enterprise
     * @return void
     */
    public function onSave(Request $request, Enterprise $enterprise): void
    {
        // Validar los datos del request
        $validatedData = $request->validate($this->rules($enterprise), $this->messages());

        // Asignar los valores de los campos validados
        $enterprise->name = $validatedData['name'];
        $enterprise->rtn = $validatedData['rtn'];
        $enterprise->email = $validatedData['email'];
        $enterprise->phone = $validatedData['phone'];
        $enterprise->address = $validatedData['address'];
        $enterprise->com = $validatedData['com'] ?? null;
        $enterprise->unit = $validatedData['unit'];
        $enterprise->price = $validatedData['price'];

        // Obtener el ID del usuario autenticado
        $userId = Auth::id();

        // Asociar el modelo al usuario autenticado si es necesario
        if (!$enterprise->exists) {  // Esto comprueba si es una creación
            $enterprise->user_id = $userId;  // Asigna directamente el user_id al modelo
        }

        // dd($request->all());

        // Procesar la carga del archivo de imagen
        if ($request->hasFile('model.upload')) {  // Verificar si el archivo fue subido
            $file = $request->file('model.upload');

            // Verificar si no hay errores en la carga
            if ($file->getError() === UPLOAD_ERR_OK) {
                $fileName = date('Ymd_His') . '_' . uniqid() . '.' . $file->getClientOriginalExtension();
                // Guardar la imagen en el sistema de archivos (en la carpeta 'public/uploads')
                $file->storeAs('uploads', $fileName, 'public');
                // Guardar solo el nombre del archivo en la base de datos
                $enterprise->image = $fileName;
            } else {
                // Manejar el error de carga
                $enterprise->image = 'Error al cargar el archivo: ' . $file->getError();
            }
        } else {
            // Si no hay imagen, puedes decidir dejar un valor predeterminado
            $enterprise->image = 'sin_imagen.jpg';
        }

        // Guardar la información de la empresa
        $enterprise->save();

        // Mostrar mensaje de éxito
        Toast::info(message: __('Datos guardados exitosamente.'));
    }
    
    with this works to me
    
    so I am using a class EnterpriseResource extends Resource
    
    all what I done it is extends Resource
